x-pack/plugins/security_solution/common/endpoint/data_loaders/index_endpoint_fleet_actions.ts

```diff
@@ -34,6 +35,8 @@ export interface IndexedEndpointAndFleetActionsForHostResponse {
 
 export interface IndexEndpointAndFleetActionsForHostOptions {
   numResponseActions?: number;
+  alertIds?: string[];
+  overrides?: { data: { command: ResponseActionsApiCommandNames } };
```

Comment:
These options seem to really be to influence what data the generator creates. Have you thought about perhaps using a subclassed `fleetActionGenerator` instead, since it is the one responsible for generating the data?

IMO, it would be best to keep the `index**` type of scripts focused on indexing documents, rather than manipulating the data that is generated via a Generator.

Reply:
I don't fully understand this, but it seems to be a good idea 👍  I'll reach out on slack when your morning comes :)

Reply:
Alright, I think I got it, could you take another look if it's better now? :)